### DISCUSSION TITLE
Self-hosted "Profile not found"

### DISCUSSION THREAD

**Discussion by syui** [bluesky-social/atproto#2645] [Clarifications and "Why?"] [answered]
`xrpc/app.bsky.actor.getProfile` in `pds` doesn't work, do you know why?

## log

```json
pds-1                        | 
{
  "level": 50,
  "time": 1721208261085,
  "pid": 6,
  "hostname": "d99c15d24ec2",
  "name": "InvalidRequest",
  "status": 400,
  "message": "Profile not found",
  "msg": "error in xrpc method app.bsky.actor.getProfile"
}

bsky-1                       | 
{
  "level": 50,
  "time": 1721208261069,
  "pid": 7,
  "hostname": "6a75d6082678",
  "name": "xrpc-server",
  "status": 400,
  "message": "Profile not found",
  "msg": "error in xrpc method app.bsky.actor.getProfile"
}
```

## pds

pds used two src.

1. ghcr.io/bluesky-social/pds:latest
2. bluesky-social/atproto/services/pds/Dockerfile

## test

```sh
f=~/.config/atproto/token.json
token=`cat $f|jq -r .accessJwt`
did=`cat $f|jq -r .did`
handle=`cat $f|jq -r .handle`
host=`cat $f|jq -r ".didDoc.service.[].serviceEndpoint"`

get_profile=app.bsky.actor.getProfile
get_timeline=app.bsky.feed.getTimeline

# get profile (did/handle)
$ curl -sL "$host/xrpc/$get_profile?actor=$did" -H "Authorization: Bearer $token"
{"error":"InvalidRequest","message":"Profile not found"}
$ curl -sL "$host/xrpc/$get_profile?actor=$handle" -H "Authorization: Bearer $token"
{"error":"InvalidRequest","message":"Profile not found"}

# get timeline
$ curl -sL "$host/xrpc/$get_timeline" -H "Authorization: Bearer $token"
{"feed":[]}
```

**Comment by itaru2622** [comment, marked as answer]
@syui regarding my self-hosting env, I succeeded to call PDS/xrpc/app.bsky.actor.getProfile as normal, like below.

my self-hosting env is based on the main branches of bluesky-social/* asof 2024-07-17 (today's commits).
refer https://github.com/itaru2622/bluesky-selfhost-env/blob/master/README.md to test with mine env.
all docker images are also published at https://hub.docker.com/search?q=itaru2622%2Fbluesky- so you need not re-build any.
just pulling docker images and starts them, with make command.

you can check all commit hashes via https://github.com/itaru2622/bluesky-REPO/tree/asof-2024-07-17
like https://github.com/itaru2622/bluesky-atproto/tree/asof-2024-07-17 for atproto (PDS and appview and others)

```bash
curl -sL "https://$host/xrpc/app.bsky.actor.getProfile?actor=$handle" -H "Authorization: Bearer $token" | jq
{
  "did": "$did",
  "handle": "$handle",
  "associated": {
    "lists": 0,
    "feedgens": 0,
    "starterPacks": 0,
    "labeler": false
  },
  "viewer": {
    "muted": false,
    "blockedBy": false
  },
  "labels": [],
  "followersCount": 0,
  "followsCount": 0,
  "postsCount": 0
}
```

**Reply by syui** [reply]
I'm very happy to hear from you. You are up to date with the self-hosted environment.
I have a question. Was there a time between 06-26r1 and 07-17 when getprofile didn't work?
Any guesses as to why it doesn't work?
Is this a pds issue?
[google translate]

**Reply by itaru2622** [reply]
I made patches for my env (https://github.com/itaru2622/bluesky-selfhost-env/tree/master/patching), 
so those patches might be fixed the bug you met above, beyond my notice. but not sure...
almost all patches are mainly related to deployment and self-hosting.

**Reply by syui** [reply]
thank you.

**Reply by syui** [reply]
I just replaced all images with https://hub.docker.com/u/itaru2622 .
But it doesn't work.
I think it might be because of my personal environment, especially because pds is set to `syu.is` instead of sub-domain.

**Reply by itaru2622** [reply]
@syui I didn't check the getprofile before, it is the first time to call that API.
these two weeks I focused on ozone/labeller, so I don't know if there was a time when getprofile would fail.

actually, relay would not work if  indigo/pull/696 (2aeb03d) is merged in your repo.
utill applying indigo/pull/705, which I made and sent today to get success with asof 07-17.

**Reply by syui** [reply]
access social-app, it calls getprofile. I cannot edit your profile because an error occurs. I don't think this is normal behavior.
I think this issue occurred when I updated bsky.
But strangely enough, other APIs are working and can be used normally from the CLI.
However, the chat API doesn't seem to work.
> error in xrpc method chat.bsky.convo.listConvos

**Reply by itaru2622** [reply]
@syui every point of my asof-XXXX, I could sign-in social-app, so getprofile was succeeded always. 
including my asof-07-05 (timing between your interests, 06-26r1 and 07-17)

about chat/DM feature, my env also failed. its different matter than getprofile.
chat/DM feature will be updated in near feature, so I decided to missing it for a while.

**Reply by syui** [reply]
Thanks.

The network is complicated because I use cloudflare. That may be the cause. I don't use caddy.

**Reply by syui** [reply]
Sorry, I deleted postgres(db) and recreated it from scratch and getprofile started working.

**Reply by itaru2622** [reply]
sounds nice.
